Pspcl/Pspcl.Web/Controllers/PreviewController.cs

```diff
@@ -1,14 +1,125 @@
-﻿using Microsoft.AspNetCore.Authorization;
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+using Pspcl.Core.Domain;
+using Pspcl.Services.Interfaces;
+using Pspcl.Web.ViewModels;
 
 namespace Pspcl.Web.Controllers
 {
-	[Authorize]
-	public class PreviewController : Controller
-	{
-		public IActionResult Preview()
-		{
-			return View();
-		}
-	}
-}
+    [Authorize]
+    public class PreviewController : Controller
+    {
+        private readonly IStockService _stockService;
+        private readonly IMapper _mapper;
+        public PreviewController(IStockService stockService, IMapper mapper)
+        {
+            _stockService = stockService;
+            _mapper = mapper;
+        }
+        [HttpGet]
+        public IActionResult Preview()
+        {
+            var jsonResponse = TempData["StockViewModel"] as string;
```

Comment:
No null checks are there, if you find TempData null, in that case, it will throw exception.
No try catch blocks. Please do proper error handling and logging through out all flows.